mesonbuild/compilers/c.py

```diff
@@ -839,11 +843,29 @@ def get_library_naming(self, env, libtype, strict=False):
             raise AssertionError('BUG: unknown libtype {!r}'.format(libtype))
         return prefixes, suffixes
 
+    def verify_libtype(self, filename: Path, libtype: str):
+        # FIXME: shared-static and static-shared return the same for now
+        if libtype in ('shared-static', 'static-shared'):
+            return True
+        proc = subprocess.Popen(['lib', '/list', filename], stdout=subprocess.PIPE)
```

Comment:
I'm not sure how safe the assumption that `lib` is in the path is.  Perhaps the potential exception raised if it can't be found should be handled.

I guess there is no possibility of a cross-tool being required.